### PR TITLE
Detect "getter" methods and automatically set refrerence internal rv policy

### PR DESCRIPTION
**Pull request by frmdstryr**
Builds on #26 .

This adds a new property `is_getter_method` to the binder and adds a check when generating the method to automatically set the return value policy to `reference_internal` when the function returns a reference has the class has a `Set<name>` method.  

Currently pybind returns a copy that must be set to apply any changes.

Eg:

Before this change I need to do:

```python
attrs = ais_shape.Attributes()
aspect = attrs.WireAspect()
aspect.SetTypeOfLine(type_of_line)
attrs.SetWireAspect(aspect)
ais_shape.SetAttributes(attrs)
```

With this change it's just

```python
ais_shape.Attributes().LineAspect().SetTypeOfLine(type_of_line)
```

### CODE REVIEW

File: pybinder/core.py [hunk 1]
```diff
@@ -1562,6 +1566,22 @@ def is_operator(self):
...
+    @property
+    def is_getter_method(self):
+        """
+        Check if binder is a public method that returns a reference and the
+        class has an another public method public method named "Set<name>".
+
+        :return: *True* if getter, *False* otherwise.
+        :rtype: bool
+        """
+        if self.is_cxx_method and self.is_public and self.rtype.spelling.endswith("&"):
```
Comment (trelau) [inline pybinder/core.py:1578]: is there not a way to check if the return type is by reference in a more sophisticated way? does the property "is_lvalue" apply here?

File: pybinder/core.py [hunk 2]
```diff
@@ -1562,6 +1566,22 @@ def is_operator(self):
...
+            setter_name = f'Set{self.spelling}'
```
Comment (trelau) [inline pybinder/core.py:1579]: Why just "Set*" methods? What's special about them vs. any method that returns a reference?

**Review by trelau — COMMENTED** [review]
Left a few minor comments. Also, possible to write some test(s) to cover this behavior?

File: pybinder/core.py (hunk 2 above)
Comment (frmdstryr) [reply pybinder/core.py:1579]: It's an assumption that that convention (Value + SetValue) is essentially a property working with a internal member.

**Pull request merged**